File: lab1/cpp/account.cpp
#include "account.h"
#include <stdexcept>

Account::Account(int id, int initial_balance) : account_id(id), balance(initial_balance) {}

int Account::get_balance() const {
    return balance;
}

int Account::get_id() const {
    return account_id;
}

void Account::deposit(int amount) {
    if (amount < 0) {
        throw std::invalid_argument("Deposit amount must be positive");
    }
    balance += amount;
}

void Account::withdraw(int amount) {
    if (amount < 0) {
        throw std::invalid_argument("Withdraw amount must be positive");
    }
    if (balance < amount) {
        throw std::runtime_error("Insufficient funds");
    }
    balance -= amount;
}

File: lab1/account.py
class Account:
    def __init__(self, account_id, initial_balance):
        self.account_id = account_id
        self.balance = initial_balance
    
    def get_balance(self):
        return self.balance
    
    def deposit(self, amount):
        self.balance += amount
    
    def withdraw(self, amount):
        if self.balance < amount:
            raise RuntimeError("Insufficient funds")
        self.balance -= amount

